### ISSUE TITLE
Use latest version of clang if available

### ISSUE THREAD

**Issue by anakrish** [Microsoft/openenclave#1688]
The default version of clang on ubuntu 18.04 is now 8.
OE SDK seems to hard-code the use of clang-7. This may not be desirable for users.

CC: @johnkord @andschwa

**Comment by andschwa** [comment]
@anakrish Yes, it does. It's a hack to ensure we pick up clang-7 (if it's available) automatically. We could extend this hack to check for clang-8 first, but we're going to have to keep extending this hack if we want to default to clang-7. There is no officially supported way to specify a compiler order to CMake without doing it externally (I mean via environment variables).

**Comment by andschwa** [comment]
[This](https://github.com/Microsoft/openenclave/blob/7016485c730c7f93bcb18af737e8ac49b6ac00ad/CMakeLists.txt#L25) is the hack. It's logic is currently "clang-7 if installed, else revert to the default (`cc`).

We had discussed making the logic clang-7 > clang > cc, but we opted against it because we didn't wanat to stray from the defaults except in this particular case. Sounds like we might want to now.

**Comment by johnkord** [comment]
Let's move the entire project from clang-7 to clang-8 now that it is released (if it's available on 16.04). Tagging @ionutbalutoiu to coordinate on this issue regarding the LLVM 8.0.0 update in 3rdparty.

**Comment by ionutbalutoiu** [comment]
> Let's move the entire project from clang-7 to clang-8 now that it is released (if it's available on 16.04). 

Clang-8 is available in the official LLVM PPAs:
* Ubuntu 16.04
```
deb http://apt.llvm.org/xenial llvm-toolchain-xenial-8 main
```
* Ubuntu 18.04
```
deb http://apt.llvm.org/bionic llvm-toolchain-bionic-8 main
```

> Tagging @ionutbalutoiu to coordinate on this issue regarding the LLVM 8.0.0 update in 3rdparty.

Sounds good! We'll update the Ansible scripts and test the `clang-8` bump with some ephemeral nodes, in order to avoid conflicts with the jobs running on the live system with `clang-7`.

**Comment by johnkord** [comment]
We may want to hold off on this to make sure that alpine Linux will be supported with this compiler as well.
